examples/k8s/scale_server/README.md

```diff
@@ -0,0 +1,63 @@
+# Scaling SPIRE server
+
+## Configuration
+
+There are a few important consideration when scaling out the number of SPIRE
+servers for availability and resiliency.
+
+### Database Persistence
+
+To scale the number of SPIRE servers, an external datasource must be used to
+maintain shared state between servers. Currently, SPIRE supports using the
+Postgres database.
+
+### Topology
+
+The SPIRE server can be deployed either stateless or stateful, however in
+either case it's imperative to configure the server appropriately to handle
+failures and scalability.
+
++ **stateless** - To run the SPIRE server stateless, the `UpstreamCA` plugin
+  needs to be used, and `upstream_bundle = true` needs to be set. An example of
+  this configuration is the [postgres example](../postgres).
```

Comment:
I'm not sure I fully understand the meaning of stateless here. The PKI state problems (e.g. signing keys etc) can be addressed by using UpstreamCA, but persistence of registration entries and some other pieces are still required?

Reply:
I mean it in the classic deployment sense where a stateless server is ephemeral, it can be created and destroyed at will with no stored state and all instances of that service are identical; where a stateful server relies on some sort of local state and require more management and consideration, such as persistent volumes or rolling in-place upgrades. In kubernetes these are first-order constructions which translate directly into a Deployment[1] and StatefulSet[2] respectively.

[1] https://kubernetes.io/docs/tasks/run-application/run-stateless-application-deployment/
[2] https://kubernetes.io/docs/tasks/run-application/run-replicated-stateful-application/

Even when using UpstreamCA, if `upstream_bundle = false` there is some server state that needs to be persisted, otherwise a loss of all SPIRE servers cannot recover without redeploying agents as the agents will not trust any of the servers.

Reply:
I see. Reading back now, I realize the entire document is geared towards running multiple SPIRE servers, and therefor shared persistence _must_ be in use. When I left this comment, I had sqlite in mind. Sorry for the confusion

Reply:
I'll see where I can be clearer about that - it **is** confusing because there's always state (persistence specifically in this case) somewhere. But yeah, this is all geared towards running multiple SPIRE servers - I'll look at it from a step back and make sure I didn't miss any other non-obvious assumptions on my part.